src/components/Disturber.tsx

```diff
@@ -0,0 +1,186 @@
+import { DrawerNavigationProp } from '@react-navigation/drawer';
+import React, { useEffect, useState } from 'react';
+import { StyleSheet, TouchableOpacity, View } from 'react-native';
+
+import { Icon, colors, normalize } from '../config';
+import { addToStore, findClosestItem, isActive, readFromStore } from '../helpers';
+import { useHomeRefresh, useStaticContent } from '../hooks';
+
+import { Button } from './Button';
+import { Image } from './Image';
+import { BoldText, RegularText } from './Text';
+import { Wrapper, WrapperHorizontal } from './Wrapper';
+
+type Props = {
+  navigation: DrawerNavigationProp<any>;
+  publicJsonFile: string;
+};
+
+interface DateRange {
+  dateEnd: string;
+  dateStart: string;
+}
+
+interface DataItem {
+  backgroundColor?: string;
+  button: {
+    navigationTo: string;
+    params: { title?: string; webUrl: string };
+    title?: string;
+  };
+  headline?: string;
+  dates: DateRange[];
+  description: string;
+  id: number;
+  picture?: { aspectRatio?: { HEIGHT: number; WIDTH: number }; uri: string };
+  title: string;
+}
+
+// eslint-disable-next-line complexity
+export const Disturber = ({ navigation, publicJsonFile }: Props) => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  const { data, refetch } = useStaticContent<DataItem[]>({
+    refreshTimeKey: `publicJsonFile-${publicJsonFile}`,
+    name: publicJsonFile,
+    type: 'json'
+  });
+
+  useHomeRefresh(refetch);
+
+  // find the closest item to the current date/time
+  const closestItem: DataItem | null = findClosestItem(
+    data?.filter((item) => item && isActive(item)) || []
+  );
+
+  const setDisturberComplete = () => {
+    setIsVisible((prev) => !prev);
+    addToStore(publicJsonFile, closestItem?.id.toString());
+  };
+
+  useEffect(() => {
+    const disturberStatus = async () => {
+      try {
+        const disturberComplete = await readFromStore(publicJsonFile);
+
+        if (closestItem?.id && disturberComplete !== closestItem?.id.toString()) {
```

Comment:
with optional chaining `disturberComplete !== closestItem?.id?.toString()` would be enough here. it will check the ability to call `id` and `toString` implicitly

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/909/commits/4dd1f20a139e88749b4a6e145ef93f008948fcab